pkg/api/types.go

```diff
@@ -486,6 +486,7 @@ type AgentPoolProfile struct {
 	DNSPrefix                           string               `json:"dnsPrefix,omitempty"`
 	OSType                              OSType               `json:"osType,omitempty"`
 	Ports                               []int                `json:"ports,omitempty"`
+	ProvisioningState                   ProvisioningState    `json:"provisioningState,omitempty"`
```

Comment:
For external consumption?

Reply:
there's already a provisioningState field in Properties, how is this different?

Reply:
you mean - for user consumption in AKS - yes.